src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationHandlers.cs

```diff
@@ -995,6 +995,9 @@ public ValueTask HandleAsync(ProcessAuthenticationContext context)
                 // Mailchimp returns the email address as a custom "login/login_email" node:
                 ProviderTypes.Mailchimp => (string?) context.UserinfoResponse?["login"]?["login_email"],
 
+                // Nextcloud returns the email address as a custom "ocs/data/email" node:
+                ProviderTypes.Nextcloud => (string?) context.UserinfoResponse?["ocs"]?["data"]?["email"],
```

Comment:
Hum, is the userinfo response wrapped? Could you please share a sample of a userinfo response for Nextcloud? (feel free to redact personal details)

Reply:
I found the answer to that question in the Nextcloud OpenAPI docs (they do wrap the userinfo details under ocs/data): https://github.com/nextcloud/server/blob/c5f72f0761df8b4e2e8e5c7c3be0bf3a3d17ad55/apps/provisioning_api/openapi.json#L2374-L2388

I pushed a commit that unwraps the userinfo response (which requires changing the claims mapping a bit).

Please give it a final try. Once you confirm it works fine, I'll merge this PR and make sure it's included in the release coming next week.

Thanks!

Reply:
Note: I documented the unwrapping process here: https://documentation.openiddict.com/guides/contributing-a-new-web-provider.html#unwrap-userinfo-responses-if-necessary.